pkg/dockerregistry/server/auditblobstore.go

```diff
@@ -0,0 +1,67 @@
+package server
+
+import (
+	"net/http"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+
+	"github.com/openshift/origin/pkg/dockerregistry/server/audit"
+)
+
+// auditBlobStore wraps a distribution.BlobStore to track operation result and
+// write it in the audit log.
+type auditBlobStore struct {
+	store distribution.BlobStore
+}
+
+var _ distribution.BlobStore = &auditBlobStore{}
+
+func (b *auditBlobStore) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
+	desc, err := b.store.Stat(ctx, dgst)
+	audit.GetLogger(ctx).Print(err, "BlobStore.Stat")
```

Comment:
All the audit logging currently logs only errors or success of operations which happens after the operation completes. Some operations may result in a panic - causing them not to be logged at all if we log just post-operation.

What I'm suggesting is to make another log entry just before the operation starts - so we have an audit entry even for those operations that panicked.